DigitalLearningSolutions.Web/Controllers/LearningPortalController/RecommendedLearningController.cs

```diff
@@ -0,0 +1,211 @@
+
+    [Authorize(Policy = CustomPolicies.UserOnly)]
+    [ServiceFilter(typeof(VerifyDelegateUserCanAccessSelfAssessment))]
+    public class RecommendedLearningController : Controller
+    {
+        private readonly IConfiguration configuration;
+        private readonly IFilteredApiHelperService filteredApiHelperService;
+        private readonly ISelfAssessmentService selfAssessmentService;
+
+        public RecommendedLearningController(
+            IFilteredApiHelperService filteredApiHelperService,
+            ISelfAssessmentService selfAssessmentService,
+            IConfiguration configuration
+        )
+        {
+            this.filteredApiHelperService = filteredApiHelperService;
+            this.selfAssessmentService = selfAssessmentService;
+            this.configuration = configuration;
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/Results")]
+        public async Task<IActionResult> SelfAssessmentResults(int selfAssessmentId)
+        {
+            var candidateId = User.GetCandidateIdKnownNotNull();
+            
+            selfAssessmentService.SetSubmittedDateNow(selfAssessmentId, candidateId);
+            selfAssessmentService.SetUpdatedFlag(selfAssessmentId, candidateId, false);
+
+            if (!configuration.IsSignpostingUsed())
+            {
+                var filteredToken = await GetFilteredToken();
+                var profile = selfAssessmentService.GetFilteredProfileForCandidateById(selfAssessmentId, candidateId);
+                var goals = selfAssessmentService.GetFilteredGoalsForCandidateId(selfAssessmentId, candidateId)
+                    .ToList();
+                var response = await filteredApiHelperService.UpdateProfileAndGoals(filteredToken, profile, goals);
+            }
+
+            return RedirectToAction("RecommendedLearning", new { selfAssessmentId });
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/RecommendedLearning")]
+        public async Task<IActionResult> RecommendedLearning(int selfAssessmentId)
+        {
+            var candidateId = User.GetCandidateIdKnownNotNull();
+            string destUrl = "/LearningPortal/SelfAssessment/" + selfAssessmentId + "/RecommendedLearning";
+            selfAssessmentService.SetBookmark(selfAssessmentId, candidateId, destUrl);
+            selfAssessmentService.UpdateLastAccessed(selfAssessmentId, candidateId);
+
+            return configuration.IsSignpostingUsed()
+                ? ReturnSignpostingRecommendedLearningView(selfAssessmentId, candidateId)
+                : await ReturnFilteredResultsView(selfAssessmentId, candidateId);
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/Filtered/PlayList/{playListId}")]
+        public async Task<IActionResult> FilteredCompetencyPlaylist(int selfAssessmentId, string playListId)
+        {
+            if (configuration.IsSignpostingUsed())

```

Comment:
Is there a reason not to use the FeatureGate attribute here? That feels more straightforward.

Reply:
Belay, just realised this is an upside-down feature gate.